### PR TITLE
member card component

### PR DESCRIPTION
**Pull request by RealOneReal**
image

---
In scope:
Create universal member card with props: 
- first name (required)
- last name (optional)
- role (optional)
- avatar image (BASE64) (optional)
* if there is no avatar, the component should display on its place first letters of the first and last name (if there is no last name - the first and the last letter of the name)
* component should contain a "kick player" icon button
* on "kick player" icon button click shoud appear a confirmation modal windo

Out of scope:
* kick yes on confirmation window does nothing

---

did not realize modal window (wait implementation from @utrennik )

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -3,15 +3,14 @@ import { Route, Switch } from 'react-router-dom';
 import { Header } from '@components/header/header';
 import { Footer } from '@components/footer/footer';
 
-import { Main } from './pages/main/main.tsx';
-
 import '@styles/main.sass';
+import { Test } from './pages/main/test.tsx';
```
Comment (HelenBassa) [inline client/src/App.tsx]: Pages will be stored in'@components/pages'.
For test page was created '@components/pages/test-page.tsx' with component - 'TestPage'.

File: client/src/pages/main/test.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+"
```
Comment (HelenBassa) [inline client/src/pages/main/test.tsx]: Maybe try this:
// Base64 string data
`const data = 'iVBORw0KGgoAAAANSUhEUgAAAFgAAABSCAYAAADQDhNSAAAA ... etc ... '`
``avatarImage={`data:image/png;base64,${data}`}``
and string with image in base64 move to assets (for example).

**Review by HelenBassa — COMMENTED** [review]
1. Pages will be stored in'@components/pages'.
For test page was created '@components/pages/test-page.tsx' with component - 'TestPage'.
2. Maybe try this:
// Base64 string data
`const data = 'iVBORw0KGgoAAAANSUhEUgAAAFgAAABSCAYAAADQDhNSAAAA ... etc ... '`
``avatarImage={`data:image/png;base64,${data}`}``
and string with image in base64 move to assets (for example).

File: client/src/App.tsx (hunk 1 above)
Comment (utrennik) [reply client/src/App.tsx]: Please use separate folders for each component

File: client/src/components/memberCard/memberCard.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Avatar, Card, CardHeader, IconButton } from '@material-ui/core';
+import { IMemberCard } from '@models/types';
+import React from 'react';
+import './memberCard.sass';
```
Comment (utrennik) [inline client/src/components/memberCard/memberCard.tsx]: please use components/UI/ folder for separate little UI components

File: client/src/components/memberCard/memberCard.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+export const MemberCard = ({ firstName, lastName, role, avatarImage }: IMemberCard) => {
+  const avatarName = lastName
+    ? `${firstName.match(/^./i)}${lastName.match(/^./i)}`.toUpperCase()
+    : `${firstName.match(/^./i)}${firstName.match(/.$/i)}`;
```
Comment (utrennik) [inline client/src/components/memberCard/memberCard.tsx]: Please use .toUpperCase() here too

File: client/src/components/memberCard/memberCard.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  const handleDelete = (event: React.MouseEvent<HTMLButtonElement>) => {
+    console.log(event);
+  };
+
+  return (
+    <Card className="member-card">
+      <Avatar
+        className="member-card-avatar"
+        style={{ backgroundColor: '#60DABF', width: '50px', height: '50px' }}
```
Comment (utrennik) [inline client/src/components/memberCard/memberCard.tsx]: Is there any sense to use the inline styles?
Please use larger fornts for the Avatar letters (maybe 23px or something) and our $text-color for the text and icon

File: client/src/components/memberCard/memberCard.sass [hunk 6]
```diff
@@ -0,0 +1,21 @@
+
+.member-card
+  width: 280px
+  height: 76px
+  padding: 0 10px 0 10px
+  display: flex
+  column-gap: 10px
+  align-items: center
```
Comment (utrennik) [inline client/src/components/memberCard/memberCard.sass]: Please use larger fonts for the user name (i think, 20px with line-height 1.2 for the name and 12px for the role will be ok ).
Maybe there is a sense to create truncate function (if text is too long, the rest will be replaced with '...')?

File: client/src/pages/main/test.tsx (hunk 2 above)
Comment (utrennik) [reply client/src/pages/main/test.tsx]: I think, we can use for the test the simple image from the assets

File: client/src/App.tsx (hunk 1 above)
Comment (RealOneReal) [reply client/src/App.tsx]: I think delete test page when i will merge branch :)

File: client/src/pages/main/test.tsx (hunk 2 above)
Comment (RealOneReal) [reply client/src/pages/main/test.tsx]: Ok, i only want testing how it work with base64 .

File: client/src/components/memberCard/memberCard.tsx (hunk 4 above)
Comment (RealOneReal) [reply client/src/components/memberCard/memberCard.tsx]: Okay

File: client/src/components/memberCard/memberCard.tsx (hunk 5 above)
Comment (RealOneReal) [reply client/src/components/memberCard/memberCard.tsx]: I only study how work with our UI library..... In documentation this is one of ways to change styles....with my class i can`t to reassiggn styles....only with inline styles,props or with changing global theme.May be later i will find more clean solution.

**Review by HelenBassa — APPROVED** [review]

File: client/src/components/memberCard/memberCard.sass (hunk 6 above)
Comment (RealOneReal) [reply client/src/components/memberCard/memberCard.sass]: Create truncate func in @utils  ( this func add with branch issue card and correct font styles for title and subtitle)

**Review by utrennik — APPROVED** [review]

File: client/src/components/footer/footer.tsx [hunk 7]
```diff
@@ -12,7 +12,7 @@ export const Footer = () => {
         <div className="footer-github">
           <ul className="github-list">
             {team.map(({ name, link }) => (
-              <li>
+              <li key={name}>
```
Comment (aksiantsiuk) [inline client/src/components/footer/footer.tsx]: Is it possible that the `name` key will be duplicated in list? You can make it like:
```js
team.map(({ name, link }, index) => (
<li key={`${name}{$index}`}>
)
```

File: client/src/components/pages/welcome-page/welcome-page.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+        <div className="connect-container error-down">
+          <div className="action-title">
+            Connect to lobby by <span>URL</span>:
+          </div>
+          <InputButton
+            buttonText="Connect"
+            valueHandler={(value: string) => {
+              console.log(value);
+            }}
```
Comment (aksiantsiuk) [inline client/src/components/pages/welcome-page/welcome-page.tsx]: As far as I understood it is temporary solution. To not miss what you did, I'd propose next way:
```js
// TODO: connect handler with redux logic
const valueHandler={(value: string) => {}

<InputButton valueHandler={valueHandler} />
```

File: client/src/components/ui/customAvatar/customAvatar.tsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import { ICustomAvatar } from '@models/types';
+
+export const CustomAvatar = ({ lastName, firstName, avatarImage }: ICustomAvatar) => {
+  const avatarName = lastName
+    ? `${firstName.match(/^./i)}${lastName.match(/^./i)}`.toUpperCase()
+    : `${firstName.match(/^./i)}${firstName.match(/.$/i)}`.toUpperCase();
+
+  return (
+    <Avatar
+      style={{ backgroundColor: '#60DABF', width: '50px', height: '50px' }}
```
Comment (aksiantsiuk) [inline client/src/components/ui/customAvatar/customAvatar.tsx]: replace style prop with className

File: client/src/components/ui/memberCard/memberCard.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+  const nameWithoutLastName = lastName ? `${firstName} ${lastName}` : firstName;
+
+  return (
+    <Card className="member-card">
+      <CustomAvatar firstName={firstName} lastName={lastName} avatarImage={avatarImage} />
+      <CardHeader
+        titleTypographyProps={{ variant: 'h6' }}
+        style={{ padding: '0' }}
+        className="member-card-header"
```
Comment (aksiantsiuk) [inline client/src/components/ui/memberCard/memberCard.tsx]: The same

File: client/src/models/types.ts [hunk 11]
```diff
@@ -5,3 +5,48 @@ export interface IWebLink {
...
+  firstName: string;
+  /*
+   * last name of User
+   */
+  lastName?: string;
+  /*
+   * avatar image
+   */
+
+  avatarImage?: string;
```
Comment (aksiantsiuk) [inline client/src/models/types.ts]: Remove line break

File: client/src/components/footer/footer.tsx (hunk 7 above)
Comment (RealOneReal) [reply client/src/components/footer/footer.tsx]: Sorry,but i only made memberCard ^)

**Review by aksiantsiuk — APPROVED** [review]

**Pull request merged**